rust/connlib/clients/apple/src/lib.rs

```diff
@@ -86,13 +87,13 @@ impl Callbacks for CallbackHandler {
         tunnel_address_v4: Ipv4Addr,
         tunnel_address_v6: Ipv6Addr,
         dns_address: Ipv4Addr,
-    ) -> Result<(), Self::Error> {
+    ) -> Result<RawFd, Self::Error> {
         self.0.on_set_interface_config(
             tunnel_address_v4.to_string(),
             tunnel_address_v6.to_string(),
             dns_address.to_string(),
         );
-        Ok(())
+        Ok(-1)
```

Comment:
nit: I think it might be more clear if we return the value returned by the actual callback on the Swift side. There we can add a comment to the effect of "Apple does not give us the file descriptor so we send -1 back to tell connlib to find it itself"

I.e. so the FFI maintains the same signature across platforms (as much as possible)